### PR TITLE
Display JIT Allocated Heap Size

### PR DESCRIPTION
**Pull request by kunalspathak**
Added a new event `MethodJitMemoryAllocatedForCode`  in https://github.com/dotnet/runtime/pull/44030 to log the JIT 's memory request for code. This PR displays that information in PerfView.

Here is the screenshot:

image

### CODE REVIEW

File: src/PerfView/JitStats.cs
Comment (brianrob) [inline src/PerfView/JitStats.cs:163]: For most people, these events aren't going to be collected, as they're too expensive to turn on by default.  I would recommend that you make the addition of this column conditional on having the data.

File: src/TraceEvent/Computers/TraceManagedProcess.cs
Comment (brianrob) [inline src/TraceEvent/Computers/TraceManagedProcess.cs:3932]: You can just return `void` here since you don't need to do anything with the method in the caller.

File: src/TraceEvent/Computers/TraceManagedProcess.cs
Comment (brianrob) [inline src/TraceEvent/Computers/TraceManagedProcess.cs:1309]: You should not need to do anything with `_method` here.  `JitAllocComplete` did all the work to capture the allocation.

File: src/TraceEvent/Computers/TraceManagedProcess.cs
Comment (brianrob) [inline src/TraceEvent/Computers/TraceManagedProcess.cs:3932]: I would recommend calling this method something like `LogJitAllocation` or something along those lines, since it's not actually a completion operation.  `MethodComplete`, for example, is responsible for marking the JIT operation complete, which is where the complete name comes from.

File: src/TraceEvent/Computers/TraceManagedProcess.cs
Comment (brianrob) [inline src/TraceEvent/Computers/TraceManagedProcess.cs:4150]: Should we name this something like `LoaderHeapAllocSize` to be clear that these are loader heap allocations, instead of C++ heap allocations?  (If so, we should also update the various related fields.)

**Review by brianrob — COMMENTED** [review]
@kunalspathak, this is looking good.  A few comments/questions.

File: src/PerfView/JitStats.cs
Comment (kunalspathak) [reply src/PerfView/JitStats.cs:163]: Sure. What is the right way to check that? Just don't display all of this if `TotalAllocSize == 0` or have some flag somewhere in TraceManagerProcess.cs inside `LoaderHeapAllocRequest` delegate?

Comment (brianrob) [reply src/PerfView/JitStats.cs:163]: Yes, using `TotalAllocSize==0` is a good way to do it.

Comment (kunalspathak) [reply src/PerfView/JitStats.cs:163]: There is lot of conditional formatting code that needs to be turned off if we need to display this column conditionally. E.g. Adding the format for [extra column in moduleTableRow](https://github.com/microsoft/perfview/pull/1289/files#diff-02da9e6f69caee6ee8d847ae7dbceb6f9dde9c94428bf278c6cc056ab786fec7R175). I am thinking of displaying `N/A` or `-` if there is no data instead of `0`? In the help section, I will add a line saying how can someone view that data. Thoughts?

Comment (kunalspathak) [reply src/PerfView/JitStats.cs:163]: Alternatively, I will need to just add this as a last column which will be easier to turn on and off.

Comment (brianrob) [reply src/PerfView/JitStats.cs:163]: I think `N/A` would probably be OK.  The key is to make sure it's clear that the data isn't there (as opposed to 0).

**Comment by kunalspathak** [comment]
@brianrob  - I think this should be ready for review.

**Review by brianrob — APPROVED** [review]
LGTM.  I will merge this once the corresponding runtime change is merged.

**Comment by kunalspathak** [comment]
@brianrob - https://github.com/dotnet/runtime/pull/44030 is merged now.

**Pull request merged**
